include/dirac_quda.h

```diff
@@ -1477,14 +1512,39 @@ namespace quda {
     virtual QudaDiracType getDiracType() const { return QUDA_ASQTADPC_DIRAC; }
 
     virtual bool hermitian() const { return true; }
+
+    /**
+     * @brief Create the coarse staggered operator.
+     *
+     * @details Takes the multigrid transfer class, which knows
+     *          about the coarse grid blocking, as well as
+     *          having prolongate and restrict member functions,
+     *          and returns color matrices Y[0..2*dim-1] corresponding
+     *          to the coarse grid hopping terms and X corresponding to
+     *          the coarse grid "clover" term. Unike the Wilson operator,
+     *          we assume a mass normalization, not a kappa normalization.
+     *          Ultimately this routine just performs the Kahler-Dirac rotation.
+     *
+     * @param T[in] Transfer operator defining the coarse grid
+     * @param Y[out] Coarse link field
+     * @param X[out] Coarse clover field
+     * @param kappa Kappa parameter for the coarse operator (ignored, set to 1.0)
+     * @param mass Mass parameter for the coarse operator (gets explicitly built into clover)
+     * @param mu Mu parameter for the coarse operator (ignored for staggered)
+     * @param mu_factor Mu scaling factor for the coarse operator (ignored for staggered)
+     * @param allow_drop_long Whether or not we're allowed to drop the long links for small aggregation sizes
+     */
+    void createCoarseOp(GaugeField &Y, GaugeField &X, const Transfer &T, double kappa, double mass, double mu,
+                        double mu_factor, bool allow_drop_long) const;
   };
 
   // Kahler-Dirac preconditioned staggered
   class DiracImprovedStaggeredKD : public DiracImprovedStaggered
   {
 
   protected:
-    mutable cudaGaugeField *Xinv; /** inverse Kahler-Dirac matrix */
+    mutable GaugeField* Xinv; /** inverse Kahler-Dirac matrix */
+    const Dirac *parent_dirac; /** Parent Dirac operator, needed to determine if parent is pc'd or not */
```

Comment:
I think this is dangerous, to make a pointer to this Dirac operator.  While the general case will have the so-called operator always exist through an MG cycle, I think for robustness it would be better to simply make a local copy of the type of this operator, for decision making in the MG solver.

Reply:
Noted---the (sloppy) motivation was to just re-use the `dirac` member of `DiracParams`, but I could just copy in the type of `dirac` instead.

Reply:
https://github.com/lattice/quda/commit/4139ae2e3733e3f08528262b4231957df056794f